src/log/Log.cc

```diff
@@ -444,10 +445,29 @@ void Log::_flush(EntryVector& t, bool crash)
       m_journald->log_entry(e);
     }
 
+    {
+      auto [it, _] = m_recent_thread_names.try_emplace(e.m_thread, now, DEFAULT_MAX_THREAD_NAMES);
+      auto& [t, names] = it->second;
+      if (names.size() == 0 || names.front() != e.m_thread_name.data()) {
+        names.push_front(e.m_thread_name.data());
+      }
```

Comment:
This could lead to duplicates if thread id is reused and names are interleaved, but I guess we can live with that.

Reply:
Thank you, fixed.

Reply:
Yes, that's intentional. I think it's useful to see the historical names for the thread.